### PR TITLE
Replace references to RLS with rust-analyzer

### PR DESCRIPTION
**Pull request by cramertj**
Inspired by [this Reddit comment](https://www.reddit.com/r/rust/comments/rmzc9c/friendly_reminder_for_our_vscode_folks_use/hpp6apz/?utm_source=reddit&utm_medium=web2x&context=3) and [the RFC to make rust-analyzer the official LSP implementation](https://rust-lang.github.io/rfcs/2912-rust-analyzer.html#branding-how-to-talk-about-rust-analyzerrls-going-forward). I couldn't find anywhere in the RFC where it called out exactly which point we'd start advertising rust-analyzer to new users, but I think we're already at a point where it is usually what folks want. Feel free to close if this isn't wanted, or if I'm skipping a step in the plan!

This PR currently includes DO_NOT_SUBMIT placeholders indicating places where
appropriate translations are needed. I unfortunately am not familiar with any of these languages and am not familiar with the current process for requesting translations. I did notice that only some languages are translating "Rust Language Server" into "Rust <something language-specific>" today, and have only added DO_NOT_SUBMIT on those translations which were previously using something other than "Rust Language Server."

### CODE REVIEW

File: locales/en-US/tools.ftl [hunk 1]
```diff
@@ -6,7 +6,7 @@ tools-editor-support-heading = First-class editor support
 tools-editor-support-description = Whether you prefer working with code from the
         command line, or using rich graphical editors, there’s a Rust
         integration available for your editor of choice. Or you can build your
-        own using the <a href="https://github.com/rust-lang/rls">Rust Language Server</a>.
+        own using the <a href="https://github.com/rust-analyzer/rust-analyzer">Rust Analyzer</a>.
```
Comment (kirawi) [inline locales/en-US/tools.ftl:9]: This says, "the Rust Analyzer"

File: locales/fr/teams.ftl [hunk 2]
```diff
@@ -4,7 +4,7 @@ governance-team-wg-nll-description = Implémenter le nouveau « borrow checker 
 governance-team-community-survey-name = Sondages
 governance-team-community-survey-description = Mener, analyser et présenter les sondages de la communauté
 governance-team-ides-name = Équipe éditeurs et EDIs
-governance-team-ides-description = Développer les EDIs, les éditeurs, et d'autres outils tels que Racer ou RLS (Rust Language Server)
+governance-team-ides-description = Développer les EDIs, les éditeurs, et d'autres outils tels que Racer ou RLS (Rust Analyzer)
```
Comment (ekzhang) [inline locales/fr/teams.ftl:7]:
```suggestion
governance-team-ides-description = Développer les EDIs, les éditeurs, et d'autres outils tels que Racer ou Rust Analyzer
```

**Comment by matklad** [comment]
My current understanding of the situation: https://github.com/rust-lang/www.rust-lang.org/pull/1242#issuecomment-1001019377

**Comment by dralley** [comment]
@matklad What about mentioning that a transition to Rust Analyzer is in progress without necessarily making it the official recommendation and updating all the links immediately?  e.g. a notice right at the top of the page with a condensed version of exactly the same points you just mentioned.

I feel like something ought to be done.  Official or not, RLS is effectively unmaintained, there has been only a few commits in the past year which are not merely bumping dependency versions.  And there's a fairly constant stream of newbies in the Discord channels who get frustrated with RLS and then come to find out that all the people "in the know" are all using something different.  It's not a great experience.

**Comment by matklad** [comment]
This does sound much better than the status quo to me personally, but I am a wrong person to ask about this, as I think I don't have authority over what we say on the website.

**Comment by jrmuizel** [comment]
Now that https://blog.rust-lang.org/2022/02/21/rust-analyzer-joins-rust-org.html has happened is it possible to merge this?

**Comment by Manishearth** [comment]
@matklad are you okay with this being merged? I am from my side.

**Comment by matklad** [comment]
okay would be an understatement! :)

**Pull request merged**